tests/unit/include/test_derivs.cxx

```diff
@@ -0,0 +1,222 @@
+#include "gtest/gtest.h"
+
+#include "bout_types.hxx"
+#include "fft.hxx"
+#include "field3d.hxx"
+#include "test_extras.hxx"
+#include "bout/constants.hxx"
+#include "bout/deriv_store.hxx"
+#include "bout/paralleltransform.hxx"
+
+#include <algorithm>
+#include <string>
+#include <tuple>
+#include <vector>
+
+// Some basic sanity checks for the derivative kernels. Checks the
+// derivatives of sin(R) where R = {X, Y, Z} for each R
+// individually. To make this as fast as possible, we use only a
+// couple of points in the non-tested directions -- not just one
+// though, as this allows us to check we're not introducing spurious
+// variation in the other directions.
+//
+// This is one of the more complicated uses of googletest! We need to
+// all combinations of methods and directions. Unfortunately, Z has
+// one more method than X and Y -- FFT. This means we can't just use
+// the provided `Combine` to produce the Cartesian product of
+// directions and methods as the latter depends on the
+// former. Instead, we instantiate the tests separately for each
+// direction and test all the methods for that direction in that
+// instantiation.
+
+namespace {
+// These are merely sanity checks, so don't expect them to agree very well!
+constexpr BoutReal derivatives_tolerance{1.e-2};
```

Comment:
How small can we go currently?

Reply:
Smallest we can do for all the first derivatives is half this.

- `W2` fails below 5.e-3
- `C2` and `W3` fail below 2.e-4
- `S2` fails below 8.e-4
- `C4` fails below 5.e-7
- `FFT` fails below 1.e-14

We could pair the tolerances with the methods, but that would add additional complexity for what's supposed to be a basic test and I think would be more fragile.

I wouldn't be opposed to dropping the tolerance to 5.e-3, but I would be wary of doing so.

Reply:
This is used as an absolute tolerance I think -- do you know what a relative tolerance limit would look like?